### DISCUSSION TITLE
[Help]: How to make carousel A scroll carousel B (React)

### DISCUSSION THREAD

**Discussion by GrzegorzDabrowski** [davidjerleke/embla-carousel#552] [Help]
In my React project, I need to use multiple carousels and whenever I use one slider I need the other one to behave the same. For example, when I drag carousel A I need carousel B to scroll simultaneously the same way in the same time. Is it possible in embla v4?

https://codesandbox.io/embed/async-darkness-mhsn2r?fontsize=14&hidenavigation=1&theme=dark

**Comment by davidjerleke** [comment]
Hi @GrzegorzDabrowski,

Thank you for your question. Before we proceed with your question I want to know:

Are you really using v4? Because it’s very old and not supported anymore.

Best,
David

**Reply by GrzegorzDabrowski** [reply]
Hi @davidjerleke, 
Thank you for your quick response. Unfortunately yes. it's an old project and we struggle with legacy, but we do our best and it works 😅 We have this project with tight deadlines and try to... adjust. Is it possible to achieve it with v4 or new versions? (Maybe it'd be a good case to convince the client to upgrade and refactor)

**Reply by davidjerleke** [reply]
Hi @GrzegorzDabrowski,

I don’t even remember what the Embla API looked like back then 😅. There’s [this CodeSandbox](https://codesandbox.io/embed/embla-carousel-vanilla-github-issue-178-hk61x?codemirror=1) which demonstrates synchronizing a slider input with a carousel but I haven’t tried if that works if you were to downgrade the Embla version to 4 something. If it works, the same approach could be used to synchronize two carousels.

By the way, what do you mean by **behaving exactly the same**? Are carousel A identical to carousel B in terms of width, number of slides and everything else? Or do you mean that carousel B should scroll the same percentage of its scrollable length when carousel A is scrolled?

I would recommend you to consider upgrading to v8 if possible. Only then I can investigate how to achieve what you want.

Best,
David

**Reply by piotrglodek** [reply]
@davidjerleke Hi I am working together with @GrzegorzDabrowski, we consider making update embla to v8. Can we achieve this in v8? [Code sanbox with embla v8](https://codesandbox.io/embed/clever-einstein-5y83my?fontsize=14&hidenavigation=1&theme=dark).

The provided link is not reflecting what we need. When the user grabs the carousel and swipes left/right the other carousel viewport should also be transformed. How to update/add to another carousel `transform` property?


https://github.com/davidjerleke/embla-carousel/assets/33925030/a9b65d54-b372-49d5-bde7-9a027c37c472

**Reply by davidjerleke** [reply]
Hello @piotrglodek,

Welcome to this discussion. Please answer the following questions I asked in my [previous comment](https://github.com/davidjerleke/embla-carousel/discussions/552#discussioncomment-6707363) so I know what you're expecting:

> By the way, what do you mean by behaving exactly the same? Are carousel A identical to carousel B in terms of width, number of slides and everything else? Or do you mean that carousel B should scroll the same percentage of its scrollable length when carousel A is scrolled? 

Best,
David

**Reply by piotrglodek** [reply]
> behaving exactly the same?

Behavior is the same, width, numberr of slides etc. including `transform` property

**Reply by davidjerleke** [reply]
@piotrglodek thanks for the response. In that case, it's very easy to achieve. Is [this CodeSandbox](https://codesandbox.io/s/embla-v8-multiple-carousels-forked-8sxzwl?file=/src/App.tsx) what you wanted to achieve?

**Reply by piotrglodek** [reply]
Thank you, it helps a lot but I think we didn't describe exactly the behavior that we want achieve. 
Our case is that we will have more than > 2 sliders for example, "A, B, C". Each change of slider should trigger the same change to other ones. It means when C changes (slide change, grab) then A, B should behave accordingly to C. 

Do you have any idea how to connect those changes to multiple sliders using react?

**Reply by davidjerleke** [reply]
@piotrglodek that's why I asked about additional details. My time is very limited so if I have all neccesary requirements from start you will get what you want faster. I can't investigate this further now but I'll have another look when possible. With that said, you should be able to achieve what you want by modifying the code I gave you.

Best,
David

**Reply by GrzegorzDabrowski** [reply]
Thank you for being so helpful and sorry for not being precise. If, by any chance, you find some time to help us with our problem, we'd be very grateful. Thanks again ❤️
